### PR TITLE
fix(node): `fs.open` 'flags' are optional

### PR DESCRIPTION
**Pull request by peterblazejewicz**
https://github.com/nodejs/node/blob/v12.0.0/lib/fs.js#L417
https://github.com/nodejs/node/blob/v12.0.0/lib/internal/fs/promises.js#L199

Updates down to v12 (as per linked source)

/cc @louwers

Thanks!

Fixes #60473

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by peterblazejewicz** [comment]
CI failures to be address in #60475

**Review by mcollina — APPROVED** [review]
lgtm

File: types/node/v14/fs/promises.d.ts [hunk 1]
```diff
@@ -184,10 +184,17 @@ declare module 'fs/promises' {
     /**
      * Asynchronous open(2) - open and possibly create a file.
      * @param path A path to a file. If a URL is provided, it must use the `file:` protocol.
-     * @param mode A file mode. If a string is passed, it is parsed as an octal integer. If not
+     * @param [flags='r'] See `support of file system `flags``.
+     * @param [mode] A file mode. If a string is passed, it is parsed as an octal integer. If not
      * supplied, defaults to `0o666`.
      */
-    function open(path: PathLike, flags: string | number, mode?: Mode): Promise<FileHandle>;
+    function open(path: PathLike, flags: string | number | undefined, mode?: string | number): Promise<FileHandle>;
```
Comment (amcasey) [inline types/node/v14/fs/promises.d.ts:191]: I'm missing something - why not just make it optional as in v17?

Comment (peterblazejewicz) [reply types/node/v14/fs/promises.d.ts:191]: oh, just too many versions, let me do this locally.

Comment (peterblazejewicz) [reply types/node/v14/fs/promises.d.ts]: amended

**Review by mcollina — APPROVED** [review]
lgtm

**Review by LinusU — APPROVED** [review]
Nice! 👏

**Review by amcasey — APPROVED** [review]
Thanks!

**Comment by peterblazejewicz** [comment]
Ready to merge 💘

**Pull request merged**